src/java/org/jivesoftware/openfire/SessionManager.java

```diff
@@ -641,10 +641,12 @@ public void broadcastPresenceToOtherResources(JID originatingResource, Presence
         JID searchJID = new JID(originatingResource.getNode(), originatingResource.getDomain(), null);
         List<JID> addresses = routingTable.getRoutes(searchJID, null);
         for (JID address : addresses) {
-            // Send the presence of the session whose presence has changed to
-            // this other user's session
-            presence.setTo(address);
-            routingTable.routePacket(address, presence, false);
+        	if (originatingResource != address) {

```

Comment:
"If the current address is not that of the originator, send the presence." - this matches the comment, in as much as it's sending the presence of the session to this user's *other* sessions. The interesting question is whether the presence is sent to the session that generated it somewhere else, since (I think) it's meant to be.